lib/ToggleButton/ToggleButtonAccessibilitySupport.js

```diff
@@ -0,0 +1,28 @@
+var
+	kind = require('enyo/kind');
+
+/**
+* @name ToggleButtonAccessibilityMixin
+* @mixin
+*/
+module.exports = {
+
+	/**
+	* @private
+	*/
+	observers: [
+		{method: 'updateAccessibilityAttributes', path: 'value'}
+	],
+
+	/**
+	* @private
+	*/
+	updateAccessibilityAttributes: kind.inherit(function (sup) {
+		return function (was, is, prop) {
+			var enabled = !this.accessibilityDisabled;
+			sup.apply(this, arguments);
+			this.setAttribute('aria-pressed', enabled ? String(this.value) : null);
+			this.setAttribute('aria-labelledby', enabled ? this.getId() : null);
```

Comment:
To the support question, I think Enyo will ultimately need to support a broader set of accessibility tooling than moonstone necessarily would. So when support for features like `aria-hidden` and -aria-labelledby` land in the enyo/AccessibilitySupport, we should be conscious of how those features work on key platforms. We don't yet, afaik, have an supported platforms list for accessibility so which platforms would still be up for debate. In the near term, I'm okay with focusing on what webOS supports provided we are clear when we make concessions to that end so we can fix things up later.

You make a valid point regarding the overlap of label and labelledby. That's something we'll need to address. Providing a default for labelledby is a poor design choice since it would complicate determining which value to take.

That said, the issue with ToggleButton still exists. The following code is supported by Moonstone but will not work:

```
{kind: ToggleButton, components: [
  {content: 'a simple control but could be anything'}
]}
```

Before changing to use `this.$.client.getId()`, that example would have worked because the inner control isn't `aria-hidden`. Now, there's a runtime error because `this.$.client` doesn't exist. That was the point I was trying to make originally.

The purpose of adding the labelledby property was to allow the developer a way to inform the mixin which control should be the label when they've customized the components as my example above.

A simpler, and likely just as functional, solution may be to choose either `this.$.client` when it exists and otherwise use the ToggleButton's id. That still puts the reasonably responsibility on the developer to correctly tag any internal controls for accessibility without overcomplicating the logic here.

```
	updateAccessibilityAttributes: kind.inherit(function (sup) {
		return function () {
			var enabled = !this.accessibilityDisabled,
				id = this.$.client ? this.$.client.id : this.id;
			sup.apply(this, arguments);
			this.setAttribute('aria-labelledby', enabled ? id : null);
		};
	})
```

```
// the developer would then need to choose which internal contents would be in the a11y tree.
{kind: ToggleButton, components: [
  {accessibilityDisabled: true, content: 'Shouldn\'t be in the accessibility tree'},
  {content: 'Relevant Content for Accessibility'}
]}
```